api/v1alpha1/envoygateway_types.go

```diff
@@ -351,6 +354,29 @@ type RateLimitMetricsPrometheusProvider struct {
 	Disable bool `json:"disable,omitempty"`
 }
 
+type RateLimitTracing struct {
+	// SamplingRate controls the rate at which traffic will be
+	// selected for tracing if no prior sampling decision has been made.
+	// Defaults to 100, valid values [0-100]. 100 indicates 100% sampling.
+	// +optional
+	SamplingRate *uint32 `json:"samplingRate,omitempty"`
+
+	// BackendRef defines the target trace collector endpoint configuration
+	BackendRef gwapiv1.BackendObjectReference `json:"backendRef"`
+
+	// AppProtocol defines the protocol of provider port in tracing feature.
+	// Only "http"(default) and "grpc" are allowed in this field
+	// +optional
+	AppProtocol *string `json:"appProtocol"`
+
+	// ClusterDomain is an optional field that specifies the custom domain used for the Kubernetes cluster.
+	// This field is used when the cluster is configured with a custom DNS domain,
+	// different from the default "cluster.local".
+	// Envoy Gateway uses this custom domain to generate fully qualified domain names (FQDN) for trace gatherer services.
+	// +optional
+	ClusterDomain string `json:"clusterDomain,omitempty"`
```

Comment:
lets rm this, not needed, we can solve this in a follow up, by inspecting backendRef in the gateway api layer